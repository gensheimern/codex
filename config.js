module.exports = {
	DB_HOST: process.env.DB_HOST || '159.65.115.221',
	DB_USER: process.env.DB_USER || 'root',
	DB_PASS: process.env.DB_PASS || 'teamcodex',
	DB_NAME: process.env.DB_NAME || 'lunch_planner',
	JWT_SECRET: process.env.JWT_SECRET || 'secret',
	MAIL_HOST: process.env.MAIL_HOST || '159.65.115.221',
	MAIL_USER: process.env.MAIL_USER || 'support@codex-team.de',
	MAIL_PASS: process.env.MAIL_PASS || 'teamcodex123',
};
