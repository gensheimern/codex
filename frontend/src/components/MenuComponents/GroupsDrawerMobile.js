import React from 'react';
import Drawer from 'material-ui/Drawer';
import FlatButton from 'material-ui/FlatButton';
import Groups from '../Groups';
import "./sidebars.css";

import IconGroup from 'material-ui/svg-icons/social/group';

export default class GroupsDrawer extends React.Component {

  constructor(props) {
    super(props);
    this.state = {open: false};
  }

  handleToggle = () => this.setState({open: !this.state.open});

  handleClose = () => this.setState({open: false});

  render() {
    console.log(this.props.searchFilterFeed)

    return (
      <div className="GroupsDrawer">
        <FlatButton
          backgroundColor="#1EA185"
          icon={ <IconGroup color="#FFFFFF"/> }
          onClick={this.handleToggle}
        />
        <Drawer
          docked={false}
          width={"84%"}
          open={this.state.open}
          onRequestChange={(open) => this.setState({open})}
        >
        <Groups changeContent={this.props.changeContent}
          searchFilterFeed={this.props.searchFilterFeed}
          closeDrawer={this.handleClose} id="groups-wrapper" height="100%" />
        </Drawer>
      </div>
    );
  }
}