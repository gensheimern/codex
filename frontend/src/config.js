const config = {
	debug: process.env.REACT_APP_DEBUG === 'TRUE',
	apiPath: process.env.REACT_APP_API_PATH,
	wsPath: process.env.REACT_APP_WS_PATH,
	basePath: process.env.REACT_APP_BASE_PATH,
}

export default config;