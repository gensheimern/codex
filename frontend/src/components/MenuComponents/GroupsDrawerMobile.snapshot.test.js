import React from 'react';
import renderer from'react-test-renderer';
import GroupsDrawerMobile from './GroupsDrawerMobile';
import config from '../../config';
import {MemoryRouter} from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
const fetchMock = require('fetch-mock');

beforeEach(() => {
    window.localStorage = {getItem: () => ""};
    fetchMock.get(config.apiPath +'/team', 
    [ { id: 6,
		name: "DÃ¶ner",
		manager: {
			id: 6,
		    firstName: "Max",
		    name: "Mustermann",
		    email: "max.mustermann@web.de",
		    image: "",
		    me: true,
		},
    }]);
});  
describe('GroupsSidebarButton Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <MemoryRouter>
                <MuiThemeProvider>
                    <GroupsDrawerMobile/>
                </MuiThemeProvider>
            </MemoryRouter>
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});