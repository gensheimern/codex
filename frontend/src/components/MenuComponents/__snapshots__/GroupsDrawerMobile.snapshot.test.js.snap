// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupsSidebarButton Snapshot renders 1`] = `
Array [
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    style={
        Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "#1EA185",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "float": "left",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": "100%",
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": "56px",
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "width": "56px",
          }
    }
    tabIndex={0}
    type="button"
>
    <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
    >
        <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
                Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "#FFFFFF",
                    "height": 24,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "verticalAlign": "middle",
                    "width": 24,
                  }
            }
            viewBox="0 0 24 24"
        >
            <path
                d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
        </svg>
    </div>
</button>,
  <div
    className={undefined}
    style={undefined}
>
    <div
        className={undefined}
        onClick={[Function]}
        style={
            Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "backgroundColor": "rgba(0, 0, 0, 0.54)",
                "height": "calc(100% - 56px)",
                "left": "-100%",
                "marginTop": "56px",
                "muiPrepared": true,
                "opacity": 0,
                "pointerEvents": "none",
                "position": "fixed",
                "top": 0,
                "transform": "translateZ(0)",
                "transition": "left 0ms cubic-bezier(0.23, 1, 0.32, 1) 400ms, opacity 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
                "willChange": "opacity",
                "zIndex": 1200,
              }
        }
    />
    <div
        className={undefined}
        style={
            Object {
                "WebkitOverflowScrolling": "touch",
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "backgroundColor": "#ffffff",
                "borderRadius": "0px",
                "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                     0 3px 10px rgba(0, 0, 0, 0.23)",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "Roboto, sans-serif",
                "height": "calc(100% - 58px)",
                "left": 0,
                "marginTop": "56px",
                "muiPrepared": true,
                "overflow": "auto",
                "position": "fixed",
                "top": 0,
                "transform": "translate(-870.16px, 0)",
                "transition": "transform 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": 860.16,
                "zIndex": 1300,
              }
        }
    >
        <div
            className="leftContent"
        >
            <div>
                <div
                    className="GroupSidebarButton"
                >
                    <button
                        className="highlightSidebarContent"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onTouchStart={[Function]}
                        style={
                            Object {
                                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                                "backgroundColor": "rgba(0, 0, 0, 0)",
                                "border": 10,
                                "borderRadius": 2,
                                "boxSizing": "border-box",
                                "color": "white",
                                "cursor": "pointer",
                                "display": "inline-block",
                                "fontFamily": "Roboto, sans-serif",
                                "fontSize": "inherit",
                                "fontWeight": "inherit",
                                "height": 36,
                                "lineHeight": "36px",
                                "margin": "0px",
                                "minWidth": "0px",
                                "muiPrepared": true,
                                "outline": "none",
                                "overflow": "hidden",
                                "padding": 0,
                                "position": "relative",
                                "textAlign": "left",
                                "textDecoration": "none",
                                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                "userSelect": "none",
                                "verticalAlign": null,
                                "width": "100%%",
                              }
                        }
                        tabIndex={0}
                        target="_blank"
                        type="button"
                    >
                        <div
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchStart={[Function]}
                        >
                            PUBLIC
                            <svg
                                onMouseEnter={[Function]}
                                onMouseLeave={[Function]}
                                style={
                                    Object {
                                        "color": "rgba(0, 0, 0, 0.87)",
                                        "display": "inline-block",
                                        "fill": "white",
                                        "float": "left",
                                        "height": 24,
                                        "marginLeft": "6%",
                                        "marginRight": "3%",
                                        "marginTop": "2%",
                                        "muiPrepared": true,
                                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                        "userSelect": "none",
                                        "verticalAlign": "middle",
                                        "width": 24,
                                      }
                                }
                                viewBox="0 0 24 24"
                            >
                                <path
                                    d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
                                />
                            </svg>
                        </div>
                    </button>
                </div>
                <div
                    className="divider"
                />
                <div
                    className="GroupSidebarButton"
                >
                    <button
                        className="nonHighlightSidebarContent"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onTouchStart={[Function]}
                        style={
                            Object {
                                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                                "backgroundColor": "rgba(0, 0, 0, 0)",
                                "border": 10,
                                "borderRadius": 2,
                                "boxSizing": "border-box",
                                "color": "white",
                                "cursor": "pointer",
                                "display": "inline-block",
                                "fontFamily": "Roboto, sans-serif",
                                "fontSize": "inherit",
                                "fontWeight": "inherit",
                                "height": 36,
                                "lineHeight": "36px",
                                "margin": "0px",
                                "minWidth": "0px",
                                "muiPrepared": true,
                                "outline": "none",
                                "overflow": "hidden",
                                "padding": 0,
                                "position": "relative",
                                "textAlign": "left",
                                "textDecoration": "none",
                                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                "userSelect": "none",
                                "verticalAlign": null,
                                "width": "100%%",
                              }
                        }
                        tabIndex={0}
                        target="_blank"
                        type="button"
                    >
                        <div
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchStart={[Function]}
                        >
                            PERSONAL
                            <svg
                                onMouseEnter={[Function]}
                                onMouseLeave={[Function]}
                                style={
                                    Object {
                                        "color": "rgba(0, 0, 0, 0.87)",
                                        "display": "inline-block",
                                        "fill": "white",
                                        "float": "left",
                                        "height": 24,
                                        "marginLeft": "6%",
                                        "marginRight": "3%",
                                        "marginTop": "2%",
                                        "muiPrepared": true,
                                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                        "userSelect": "none",
                                        "verticalAlign": "middle",
                                        "width": 24,
                                      }
                                }
                                viewBox="0 0 24 24"
                            >
                                <path
                                    d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
                                />
                            </svg>
                        </div>
                    </button>
                </div>
                <div
                    className="divider"
                />
                <div
                    style={
                        Object {
                            "width": "100%",
                          }
                    }
                >
                    <p
                        style={
                            Object {
                                "color": "#ffffff",
                                "float": "left",
                                "margin": "5%",
                                "marginLeft": "10%",
                                "marginRight": "2%",
                                "marginTop": "3%",
                                "textAlign": "left",
                              }
                        }
                    >
                        GROUPS
                    </p>
                    <div>
                        <div
                            className="CreateTeamButton"
                        >
                            <button
                                disabled={false}
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onKeyUp={[Function]}
                                onMouseEnter={[Function]}
                                onMouseLeave={[Function]}
                                onTouchStart={[Function]}
                                style={
                                    Object {
                                        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                                        "backgroundColor": "rgba(0, 0, 0, 0)",
                                        "border": 10,
                                        "borderRadius": 2,
                                        "boxSizing": "border-box",
                                        "color": "white",
                                        "cursor": "pointer",
                                        "display": "inline-block",
                                        "fontFamily": "Roboto, sans-serif",
                                        "fontSize": "inherit",
                                        "fontWeight": "inherit",
                                        "height": 36,
                                        "lineHeight": "36px",
                                        "margin": "0px",
                                        "minWidth": "0px",
                                        "muiPrepared": true,
                                        "outline": "none",
                                        "overflow": "hidden",
                                        "padding": 0,
                                        "position": "relative",
                                        "textAlign": "center",
                                        "textDecoration": "none",
                                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                        "userSelect": "none",
                                        "verticalAlign": null,
                                      }
                                }
                                tabIndex={0}
                                target="_blank"
                                type="button"
                            >
                                <div
                                    onMouseDown={[Function]}
                                    onMouseLeave={[Function]}
                                    onMouseUp={[Function]}
                                    onTouchEnd={[Function]}
                                    onTouchStart={[Function]}
                                >
                                    <svg
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        style={
                                            Object {
                                                "color": "rgba(0, 0, 0, 0.87)",
                                                "display": "inline-block",
                                                "fill": "#FFFFFF",
                                                "height": 24,
                                                "marginLeft": 0,
                                                "marginRight": 0,
                                                "muiPrepared": true,
                                                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                                "userSelect": "none",
                                                "verticalAlign": "middle",
                                                "width": 24,
                                              }
                                        }
                                        viewBox="0 0 24 24"
                                    >
                                        <path
                                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                                        />
                                    </svg>
                                </div>
                            </button>
                        </div>
                    </div>
                    <div
                        className="groups"
                    />
                </div>
            </div>
        </div>
    </div>
</div>,
]
`;
